### PR TITLE
Auto-delete database and create lao tests

### PR DESCRIPTION
**Pull request by BadrTad**
+ Delete database (for scala) after each scenario.
+ Tests for create LAO:
    > empty name test
    > negative time
    > invalid id
+ Data test was placed in src/test/java/data/lao in order to be used in karate

### CODE REVIEW

File: tests/karate/src/test/java/be/utils/GoServer.java [hunk 1]
```diff
@@ -28,4 +28,11 @@ public String getDir() {
   public String getLogPath() {
     return Paths.get("go.log").toString();
   }
+
+  @Override
+  public void deleteDatabaseDir() {
+    // TODO Auto-generated method stub
+
```
Comment (GabrielFleischer) [inline tests/karate/src/test/java/be/utils/GoServer.java:35]: Maybe add a real comment here

```suggestion
    // The Go backend does not need to delete the database
```

File: tests/karate/src/test/java/be/utils/ScalaServer.java [hunk 2]
```diff
@@ -29,4 +37,36 @@ public String getDir() {
...
+        Thread.sleep(1500);
+        i++;
+      }
+      catch(InterruptedException e){
+        System.err.println("Delete thread of file " + file.toPath() + " was interrupted");
+      }
+    }
+    if(i >= Configurable.MAX_DB_DELETE_ATTEMPTS){
+      System.err.println("Could not delete database folder" + file.toPath());
+    }
```
Comment (GabrielFleischer) [inline tests/karate/src/test/java/be/utils/ScalaServer.java:70]: Formatting issues
```suggestion
  private static void deleteHandled(File file) {
    int i = 0;
    while (i < Configurable.MAX_DB_DELETE_ATTEMPTS && !file.delete()) {
      try {
        //Wait for the server to realease the resource(database)
        //and try again
        Thread.sleep(1500);
        i++;
      } catch(InterruptedException e) {
        System.err.println("Delete thread of file " + file.toPath() + " was interrupted");
      }
    }
    if (i >= Configurable.MAX_DB_DELETE_ATTEMPTS) {
      System.err.println("Could not delete database folder" + file.toPath());
    }
```

**Review by GabrielFleischer — COMMENTED** [review]
A chunk of the code should be reformatted. But otherwise, this look great !

File: tests/karate/src/test/java/be/utils/ScalaServer.java [hunk 3]
```diff
@@ -29,4 +37,36 @@ public String getDir() {
   public String getLogPath() {
     return Paths.get("scala.log").toString();
   }
+
+  @Override
+  public void deleteDatabaseDir() {
+    System.out.println("Deleting...");
```
Comment (pierluca) [inline tests/karate/src/test/java/be/utils/ScalaServer.java:43]:
```suggestion
    System.out.println("Deleting database...");
```

File: tests/karate/src/test/java/be/utils/ScalaServer.java (hunk 2 above)
Comment (BadrTad) [reply tests/karate/src/test/java/be/utils/ScalaServer.java:70]: Fixed :+1:

File: tests/karate/src/test/java/be/utils/ScalaServer.java (hunk 3 above)
Comment (BadrTad) [reply tests/karate/src/test/java/be/utils/ScalaServer.java:43]: :+1:

File: tests/karate/src/test/java/be/utils/Configurable.java [hunk 4]
```diff
@@ -16,4 +19,9 @@
    * Returns the path to output log files.
    */
   String getLogPath();
+
+  /**
+   * Deletes database of the server
+   */
+  void deleteDatabaseDir();
```
[the remark this answers is not in the document]
Comment (BadrTad) [reply tests/karate/src/test/java/be/utils/Configurable.java:26]: Agree

File: tests/karate/src/test/java/be/createLAO/create.feature [hunk 5]
```diff
@@ -43,6 +43,34 @@ Feature: Create a pop LAO
...
+    *  karate.log('Sent: '+ karate.pretty(createLaoReq))
     And   string answer = socket.listen(timeout)
     * karate.log('Received answer = ' + answer)
     Then match answer == createLaoRes
+
+  Scenario: Create Lao request with empty lao name should fail with an error response
+    Given string  emptyNameReq = read('classpath:data/lao/bad_lao_create_empty_name.json')
+    And   def socket = karate.webSocket(wsURL,handle)
+    When  eval socket.send(emptyNameReq)
+    *  karate.log('Sent: '+ karate.pretty(emptyNameReq))
```
[the remark this answers is not in the document]
Comment (BadrTad) [reply tests/karate/src/test/java/be/createLAO/create.feature:55]: :+1:

File: tests/karate/src/test/java/be/utils/ScalaServer.java [hunk 6]
```diff
@@ -17,7 +25,7 @@ public void stop() {
   @Override
   public String[] getCmd() {
     String configPath = Paths.get("src", "main", "scala", "ch", "epfl", "pop", "config").toString();
-    return new String[]{"sbt", "-Dscala.config=" + configPath, "run"};
+    return new String[] { "sbt.bat", "-Dscala.config=" + configPath, "run" };
```
Comment (GabrielFleischer) [inline tests/karate/src/test/java/be/utils/ScalaServer.java:28]: Isn't that windows specific ?

Are you sure we can run this in a linux environment ?

**Review by GabrielFleischer — APPROVED** [review]
I am sorry for the time it took me to review this.

I just have one fear about the .bat change.
I am not sure if it will work in a Linux environment.

File: tests/karate/src/test/java/be/utils/ScalaServer.java (hunk 6 above)
Comment (pierluca) [reply tests/karate/src/test/java/be/utils/ScalaServer.java:28]: Agreed, this (most likely) won't work in a linux environment.

**Review by GabrielFleischer — APPROVED** [review]

**Pull request merged**